.github/workflows/build.yml

```diff
@@ -0,0 +1,71 @@
+name: Build Image
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Maximize Build Space
+      uses: easimon/maximize-build-space@master
+      with:
+        root-reserve-mb: 10000 
+        swap-size-mb: 1024
+        remove-dotnet: 'true'
+        remove-android: 'true'
+        remove-haskell: 'true'
+        remove-codeql: 'true'
+        remove-docker-images: 'true'
+
+    - name: List Available Space
+      run: |
+        echo "Free space:"
+        df -h
+
+    - name: Install Dependencies
```

Comment:
You could reuse the [declared build dependencies](https://github.com/RPi-Distro/pi-gen/blob/master/depends) instead of listing them manually.

Reply:
Good call.  Mostly using `depends` now, but there was one additional needed dependency, `qemu-user-static`.  Not sure whether the right thing is to put it into `depends`, but for now it's installed separately as part of the build step.

Reply:
Maybe @XECDesign can elaborate on whether this could be added to `depends` as well.